src/Microsoft.AspNetCore.Authentication.OpenIdConnect/OpenIdConnectHandler.cs

```diff
@@ -585,6 +578,11 @@ protected virtual Task<bool> HandleSignOutCallbackAsync()
                         tokenEndpointResponse = await RedeemAuthorizationCodeAsync(tokenEndpointRequest);
                     }
 
+                    if (!string.IsNullOrEmpty(tokenEndpointResponse.Error))
```

Comment:
Yeah, forget that, `TokenResponseReceived` is not invoked for error responses due to the `EnsureSuccessStatusCode` call, so no regression here :smile: 

> How to set status code for failed authentication response is not defined in the spec.

Actually, it is: https://tools.ietf.org/html/rfc6749#section-5.2